### DISCUSSION TITLE
How to integrate with the Agoric AMM?

### DISCUSSION THREAD

**Discussion by dckc** [Agoric/agoric-sdk#5644] [Q&A]
@kennyrowe your suggestion is a good one:

 - [Arbitrage Bot · Issue \#4656 · Agoric/agoric\-sdk](https://github.com/Agoric/agoric-sdk/issues/4656)

another example is:
 - [amm_swap.js](https://github.com/Agoric/instagoric/blob/main/bases/ag-solo-tasks/tasks/amm_swap.js)

starting from zero, there's also:
 - [How to Install the Agoric SDK \- YouTube](https://www.youtube.com/watch?v=VrXlGwKh0zk)
 - [Getting started: installing the Agoric SDK](https://agoric.com/documentation/getting-started/before-using-agoric.html)

cc @rowgraus @btulloh

**Comment by dckc** [comment]
see also some rough plans in this area: https://github.com/Agoric/agoric-sdk/issues/5663

**Comment by dckc** [comment]
recent concrete progress in #5663:
 - [swap-on-amm.js](https://gist.github.com/dckc/73205178bd44cf3f4bbcf7db0fc280a8#file-swap-on-amm-js)
 - [borrow-from-vault.js](https://gist.github.com/dckc/73205178bd44cf3f4bbcf7db0fc280a8#file-borrow-from-vault-js)

these are from live-coding sessions. Recordings are available: https://github.com/Agoric/documentation/issues/576#issuecomment-1177914880

**Comment by dckc** [comment]
https://agoric.com/documentation/zoe/guide/contracts/constantProductAMM.html  is our normal documentation

@Chris-Hibbert suggests https://github.com/Agoric/agoric-sdk/blob/master/packages/run-protocol/src/vpool-xyk-amm/constantProduct/README.md  for more detail.

**Comment by Tartuffo** [comment]
Issue for this: https://github.com/Agoric/documentation/issues/616, assigned to @cboydstun
